examples/docs_beta_snippets/docs_beta_snippets_tests/snippet_checks/guides/components/test_components_docs_creating_a_component.py

```diff
@@ -32,27 +33,7 @@
 def test_components_docs_index(
     update_snippets: bool, update_screenshots: bool, get_selenium_driver
 ) -> None:
-    snip_no = 0
-
-    def next_snip_no():
-        nonlocal snip_no
-        snip_no += 1
-        return snip_no
-
-    with (
-        TemporaryDirectory() as tempdir,
-        environ(
-            {
-                "COLUMNS": "90",
-                "NO_COLOR": "1",
-                "HOME": "/tmp",
-                "DAGSTER_GIT_REPO_DIR": str(DAGSTER_ROOT),
-                "VIRTUAL_ENV": "",
-            }
-        ),
-    ):
-        os.chdir(tempdir)
-
+    with isolated_snippet_generation_environment() as get_next_snip_number:
```

Comment:
this is an odd API but i'm not going to sweat too much about it given the test-only nature.. I think you could consider pulling the get_next_snip_number stuff into its own separate thing though